packages/messaging/src/models/errors.ts

```diff
@@ -17,139 +17,142 @@
 
 import { ErrorFactory } from '@firebase/util';
 
-export const ERROR_CODES = {
-  AVAILABLE_IN_WINDOW: 'only-available-in-window',
-  AVAILABLE_IN_SW: 'only-available-in-sw',
-  SHOULD_BE_INHERITED: 'should-be-overriden',
-  BAD_SENDER_ID: 'bad-sender-id',
-  INCORRECT_GCM_SENDER_ID: 'incorrect-gcm-sender-id',
-  PERMISSION_DEFAULT: 'permission-default',
-  PERMISSION_BLOCKED: 'permission-blocked',
-  UNSUPPORTED_BROWSER: 'unsupported-browser',
-  NOTIFICATIONS_BLOCKED: 'notifications-blocked',
-  FAILED_DEFAULT_REGISTRATION: 'failed-serviceworker-registration',
-  SW_REGISTRATION_EXPECTED: 'sw-registration-expected',
-  GET_SUBSCRIPTION_FAILED: 'get-subscription-failed',
-  INVALID_SAVED_TOKEN: 'invalid-saved-token',
-  SW_REG_REDUNDANT: 'sw-reg-redundant',
-  TOKEN_SUBSCRIBE_FAILED: 'token-subscribe-failed',
-  TOKEN_SUBSCRIBE_NO_TOKEN: 'token-subscribe-no-token',
-  TOKEN_SUBSCRIBE_NO_PUSH_SET: 'token-subscribe-no-push-set',
-  TOKEN_UNSUBSCRIBE_FAILED: 'token-unsubscribe-failed',
-  TOKEN_UPDATE_FAILED: 'token-update-failed',
-  TOKEN_UPDATE_NO_TOKEN: 'token-update-no-token',
-  USE_SW_BEFORE_GET_TOKEN: 'use-sw-before-get-token',
-  INVALID_DELETE_TOKEN: 'invalid-delete-token',
-  DELETE_TOKEN_NOT_FOUND: 'delete-token-not-found',
-  DELETE_SCOPE_NOT_FOUND: 'delete-scope-not-found',
-  BG_HANDLER_FUNCTION_EXPECTED: 'bg-handler-function-expected',
-  NO_WINDOW_CLIENT_TO_MSG: 'no-window-client-to-msg',
-  UNABLE_TO_RESUBSCRIBE: 'unable-to-resubscribe',
-  NO_FCM_TOKEN_FOR_RESUBSCRIBE: 'no-fcm-token-for-resubscribe',
-  FAILED_TO_DELETE_TOKEN: 'failed-to-delete-token',
-  NO_SW_IN_REG: 'no-sw-in-reg',
-  BAD_SCOPE: 'bad-scope',
-  BAD_VAPID_KEY: 'bad-vapid-key',
-  BAD_SUBSCRIPTION: 'bad-subscription',
-  BAD_TOKEN: 'bad-token',
-  BAD_PUSH_SET: 'bad-push-set',
-  FAILED_DELETE_VAPID_KEY: 'failed-delete-vapid-key',
-  INVALID_PUBLIC_VAPID_KEY: 'invalid-public-vapid-key',
-  USE_PUBLIC_KEY_BEFORE_GET_TOKEN: 'use-public-key-before-get-token',
-  PUBLIC_KEY_DECRYPTION_FAILED: 'public-vapid-key-decryption-failed'
-};
+export const enum ErrorCodes {
```

Comment:
Don't you usually give enums singular names, i.e. ErrorCode? As a type it describes what it means to be an ErrorCode value, this is no longer just an untyped bag of error codes.

Reply:
Yeah that's true, thanks. Done.